core/templates/pages/about-page/about-page.component.spec.ts

```diff
@@ -55,165 +60,175 @@ class MockI18nLanguageCodeService {
+              }
+            }
+          }
+        }
+      ]
     }).compileComponents();
-    fixture = TestBed.createComponent(AboutPageComponent);
-    component = fixture.componentInstance;
-  });
-
-  it('should click on about tab', () => {
-    component.ngOnInit();
-    expect(component.activeTabName).toBe('about');
-
-    component.onTabClick('about');
-
-    expect(windowRef.nativeWindow.location.hash).toBe('#about');
-    expect(component.activeTabName).toBe('about');
   });
+  beforeEach(angular.mock.module('oppia'));
 
-  it('should click on license tab', fakeAsync(() => {
-    component.ngOnInit();
-    expect(component.activeTabName).toBe('about');
-
-    component.onTabClick('license');
-    fixture.detectChanges();
-    expect(windowRef.nativeWindow.location.hash).toBe('#license');
-    expect(component.activeTabName).toBe('foundation');
-  }));
-
-  it('should click on foundation tab', () => {
-    component.ngOnInit();
-    expect(component.activeTabName).toBe('about');
-
-    component.onTabClick('foundation');
-
-    expect(windowRef.nativeWindow.location.hash).toBe('#foundation');
-    expect(component.activeTabName).toBe('foundation');
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
+    });
+    loaderService = TestBed.get(LoaderService);
+    userService = TestBed.get(UserService);
   });
-
-  it('should click on credits tab', () => {
-    component.ngOnInit();
-    expect(component.activeTabName).toBe('about');
-
-    component.onTabClick('credits');
-
-
-    expect(windowRef.nativeWindow.location.hash).toBe('#credits');
-    expect(component.activeTabName).toBe('credits');
+  let component;
+  beforeEach(() => {
+    const aboutPageComponent = TestBed.createComponent(AboutPageComponent);
+    component = aboutPageComponent.componentInstance;
   });
 
-  it('should activate about tab on init', () => {
-    windowRef.nativeWindow.location.hash = '#about';
-
-    component.ngOnInit();
+  it('should successfully instantiate the component from beforeEach block',
+    () => {
+      expect(component).toBeDefined();
+      component.ngOnInit();
+    });
 
-    expect(windowRef.nativeWindow.location.hash).toBe('#about');
-    expect(component.activeTabName).toBe('about');
+  it('should get static image url', () => {
+    expect(component.getStaticImageUrl('/path/to/image')).toBe(
+      '/assets/images/path/to/image');
   });
 
-  it('should activate about license on init', () => {
-    windowRef.nativeWindow.location.hash = '#license';
-
+  it('should check user is not logged in and navigate to create lesson page',
+    fakeAsync(() => {
+      const UserInfoObject = {
+        is_moderator: false,
+        is_admin: false,
+        is_super_admin: false,
+        is_topic_manager: false,
+        can_create_collections: false,
+        preferred_site_language_code: null,
+        username: '',
+        email: '',
+        user_is_logged_in: null
```

Comment:
I'm expecting this to be false*

Reply:
Here it is null because in function we have checked the condition with null. If we change it to false in function(in ts file) then it gives error.

Reply:
Can you please change the functionality such that `user_is_logged_in` is always a boolean value?

Reply:
We are checking the condition in ts file with null https://github.com/oppia/oppia/pull/11420/files#r584183470 not false as, if we check it with false then it takes us to '/_ah/login' but the login button is not clickable and we can perform no other actions. But with null it first lets the unlogged user sign in and then takes them to create mode. And as we have checked the condition with null in ts file if we check it with false in spec file it gives error so we need to check it with null only. I hope its clear now.

Reply:
The above link is not work!

Anyway, I'm not sure why we are expecting `user_is_logged_in` to be null in the backend response for a guest user?